project2.qmd

```diff
@@ -111,6 +111,119 @@ print(combined_plot)
+
+# List of selected variables based on GVIF criteria
+selected_vars <- c("carbpressure", "carbtemp", "fillerlevel", 
+                   "balling", "bowlsetpoint", "alchrel", "ballinglvl", "ph")
+
+# Filter the train data frame to only include the selected variables
+selected_data <- train[, selected_vars, drop = FALSE]
+
+# Check if the data is all numeric
+if(all(sapply(selected_data, is.numeric))) {
+  # Use chart.Correlation to create a correlation plot for the selected variables
+  chart.Correlation(selected_data, histogram=TRUE, pch=19)
+} else {
+  cat("Non-numeric columns are present in the selected data.")
+}
+```
+
+```{r message=FALSE, warning=FALSE}
+exclude_columns <- c("carbpressure", "fillerlevel", "ballinglvl", "balling", "alchrel")  
+
+# Create the model without the excluded columns
+lm_result <- lm(ph ~ ., data = train[, !colnames(train) %in% exclude_columns])
+vif_values <- vif(lm_result)
+print(vif_values)
+```
+
+Remove $carbpressure$, $fillerlevel$, $ballinglvl$, $balling$, and $alchrel$ from train and test dataframe
+
+```{r message=FALSE, warning=FALSE}
+# Remove specific columns from the train data frame
+train <- train[, !(colnames(train) %in% c("carbpressure", "fillerlevel", "ballinglvl", "balling","alchrel"))]
+
+# Remove specific columns from the predict_me data frame
+predict_me <- predict_me[, !(colnames(predict_me) %in% c("carbpressure", "fillerlevel", "ballinglvl", "balling", "alchrel"))]
+```
+
+## Missing Value
+
+### Non-linear
+
+Train dataset is a non-linear dataset. 
+```{r}
+train_complete <- na.omit(train)
+
+# Fit a linear model to the cleaned data
+model <- lm(ph ~ ., data = train_complete)
+
+# Calculate residuals and fitted values
+train_complete$residuals <- residuals(model)
+train_complete$fitted_values <- fitted(model)
+
+# Plotting the residuals against the fitted values
+ggplot(train_complete, aes(x = fitted_values, y = residuals)) +
+  geom_point(color = "slategray4") +
+  geom_hline(yintercept = 0, linetype = "dashed", color = "indianred", size = 3, linewidth = 1, boxlinewidth = 0.4) +
+  labs(x = "Fitted Values", y = "Residuals", title = "Residuals vs Fitted Values") +
+  theme_minimal()
+```
+
+### bagimpute: Bagged-Tree
+
+Fill in missing values with bagImpute. BagImpute can capture non-linear relationships among the variables in the data.
+
+```{r message=FALSE, warning=FALSE}
+sum(is.na(train))
+sum(is.na(predict_me))
+```
+
+```{r message=FALSE, warning=FALSE}
+library(caret)
+
+# Define the list of columns excluding the 'ph' column
+columns_to_include <- setdiff(names(train), "ph")
+# Temporarily store the 'ph' column
+ph_column <- train$ph
+# Remove the 'ph' column
+train$ph <- NULL
+# Use the preProcess model to fill in missing values for the remaining data
+preProc_without_ph <- preProcess(train[columns_to_include], method = "bagImpute")
+# Use the preProcess model to fill in missing values
+train <- predict(preProc_without_ph, newdata = train[columns_to_include])
```

Comment:
This doesn't work for me. All the stuff above it does, but this predict -> reassign to train doesn't work.  It's going to take me a while, but did you mean to remove/overwrite the train variables?  


preProc_without_ph <- preProcess(train[columns_to_include], method = "bagImpute")
> train <- predict(preProc_without_ph, newdata = train[columns_to_include])
Error in UseMethod("predict") : 
  no applicable method for 'predict' applied to an object of class "NULL"